File: app/src/main/java/iaruchkin/courseapp/network/TopStoriesEndpoint.java
package iaruchkin.courseapp.network;

import android.support.annotation.NonNull;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface TopStoriesEndpoint {

    @GET("topstories/v2/{section}.json")
    Single<TopStoriesResponse> get(@Path("section") @NonNull String section);
}

File: app/src/main/java/iaruchkin/courseapp/network/TopStoriesEndpoint.kt
package iaruchkin.courseapp.network

import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Path

interface TopStoriesEndpoint {

//    @GET("topstories/v2/{section}.json")
//    operator fun get(@Path("section") section: String): Single<TopStoriesResponse>

    @GET("topstories/v2/{section}.json")
    operator fun get(@Path("section") section: String): Single<String>
}


